client/utils/use-confirm-navigation.js

```diff
@@ -14,11 +14,17 @@ import { getHistory } from '@woocommerce/navigation';
  * - useConfirmNavigation( saved => { if ( ! saved ) return 'Discard unsaved changes?' }, [ saved ] );
  *
  * @param {Function} getMessage returns confirmation message string if one should appear
- * @param {Array} deps effect dependencies
+ * @return {Function} The callback to execute
  */
-const useConfirmNavigation = ( getMessage, deps ) => {
```

Comment:
Refactored to remove `deps`, because otherwise the rule couldn't determine the dependencies to `useEffect`.
Now this function returns a callback to be used within `useEffect`. And the reference to `getMessage` is "stabilized" through `useRef`.

Reply:
I guess the "Usage" above is not correct anymore?

```
 * - useConfirmNavigation( saved => { if ( ! saved ) return 'Discard unsaved changes?' }, [ saved ] );
```

Reply:
Ah, good catch, thanks! Fixed in 67db5624